public/js/graph/react_graph.js

```diff
@@ -139,8 +139,13 @@ var Graph = React.createClass({
 
     componentDidMount: function () {
         this.getGraph();
-        document.body.addEventListener('keydown', this.onKeyDown);
-        document.body.addEventListener('wheel', this.onWheel);
+        document.getElementById('react-graph').addEventListener('keydown', this.onKeyDown);
+        document.getElementById('react-graph').addEventListener('wheel', this.onWheel);
+    },
+
+    componentDidUnmount: function () {
```

Comment:
This should be `componentWillUnmount`. Obviously if it's already unmounted, there'd be no way for the component to really know that... :)

 Check out the [lifecycle docs](https://facebook.github.io/react/docs/component-specs.html) for more info.